django/src/rdwatch/views/tile.py

```diff
@@ -111,45 +114,38 @@ def satelliteimage_raster_redirect(
         or "spectrum" not in request.GET
     ):
         return HttpResponseBadRequest()
-    satimg = (
-        SatelliteImage.objects.filter(
-            constellation__slug=request.GET["constellation"],
-            spectrum__slug=request.GET["spectrum"],
-            level__slug=request.GET["level"],
-        )
-        .alias(
-            timedistance=GistDistance(
-                F("timestamp"),
-                Value(
-                    datetime.fromisoformat(str(request.GET["timestamp"])),
-                    output_field=DateTimeField(),
-                ),
-            )
-        )
-        .order_by("timedistance")
-        .first()
-    )
-    if satimg is None:
+
+    constellation = Constellation(slug=request.GET["constellation"])
+    timestamp = datetime.fromisoformat(str(request.GET["timestamp"]))
+
+    # Calculate the bounding box from the given x, y, z parameters
+    bounds = mercantile.bounds(x, y, z)
+    bbox = (bounds.west, bounds.south, bounds.east, bounds.north)
+
+    bands = get_bands(constellation, timestamp, bbox)
+
+    try:
+        band: Band = next(bands)
```

Comment:
Rather than grabbing the first band available, we should get the band filtered by:

- the requested processing level: `level`
- the requested spectrum (common band name): `spectrum`

From that filtered list, we should then sort it by:
- the closest timestamp to the requested timestamp. Since we allow a time "buffer" in our STAC search, it's possible (but unlikely) there will be multiple images returned within that buffer.
- preferring URIs with the ".tif" extension. For Sentinel 2, there's the possibility of getting a JP2000 or a Cloud Optimized GeoTIFF encoding for the exact same image. Cloud Optimized GeoTIFF is both cheaper and faster to generate tiles for.